File: C#/routes/Web.cs
using Csharp.Controllers;

namespace Csharp.routes;

public static class Web
{
    public static Dictionary<string, (object controller, System.Reflection.MethodInfo method)> Routes =
        new()
        {
            { "/", (new UserController(), typeof(UserController).GetMethod("Index")!) },
            { "/home", (new HomeController(), typeof(HomeController).GetMethod("Index")!) },
            { "/api/users", (new UserController(), typeof(UserController).GetMethod("Index")!) },
        };
}

File: go/routes/web.go
package routes

import (
	"project/src/Controllers"
)

type Route struct {
	Controller interface{}
	Method     func(interface{}) interface{}
}

var Routes = map[string]Route{
	"/": {
		Controller: &Controllers.UserController{},
		Method: func(c interface{}) interface{} {
			return c.(*Controllers.UserController).Index()
		},
	},
	"/home": {
		Controller: &Controllers.HomeController{},
		Method: func(c interface{}) interface{} {
			return c.(*Controllers.HomeController).Index()
		},
	},
	"/api/users": {
		Controller: &Controllers.UserController{},
		Method: func(c interface{}) interface{} {
			return c.(*Controllers.UserController).Index()
		},
	},
}
